src/Parser/RestoDruid/Modules/Features/StatWeights.js

```diff
@@ -0,0 +1,256 @@
+import HealingValue from 'Parser/Core/Modules/HealingValue';
+import CritEffectBonus from 'Parser/Core/Modules/Helpers/CritEffectBonus';
+
+import { getSpellInfo } from '../../SpellInfo';
+
+const DEBUG = true;
+
+const ARMOR_INT_MULTIPLIER = 1.05; // 5% int bonus from wearing all leather means each new point of int worth 1.05 vs character sheet int
+
+class StatWeights extends Module {
+  static dependencies = {
+    combatants: Combatants,
+    critEffectBonus: CritEffectBonus,
+  };
+
+  concordanceAmount = 0; // TODO remove when stat tracker added
+
+  totalAdjustedHealing = 0; // total healing after excluding 'multiplier' spells like Leech / Velens
+
+  // These are the total healing that would be gained if their respective stats ratings were increased by one.
+  totalOneInt = 0;
+  totalOneCrit = 0;
+  totalOneHasteHpm = 0;
+//  totalOneHasteHpct = 0;
+  totalOneMastery = 0;
+  totalOneVers = 0;
+  totalOneLeech = 0;
+
+  on_initialized() {
+    this.concordanceAmount = 4000 + (this.combatants.selected.traitsBySpellId[SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_TRAIT.id] || 0) * 300; // TODO remove when stat tracker added
+  }
+
+  on_byPlayer_heal(event) {
+    const healVal = new HealingValue(event.amount, event.absorbed, event.overheal);
+    this._handleHeal(event, healVal);
+  }
+
+  on_byPlayer_absorbed(event) {
+    const healVal = new HealingValue(event.amount, 0, 0);
+    this._handleHeal(event, healVal);
+  }
+
+  _handleHeal(event, healVal) {
+    const target = this.combatants.getEntity(event);
+    if(target === null) {
+      return;
+    }
+
+    const amount = healVal.effective;
+    const spellInfo = getSpellInfo(event.ability.guid);
+
+    // LEECH //
+    // We have to calculate leech weight differently depending on if we already have any leech rating.
+    // Leech is marked as a 'multplier' heal, so we have to check it before we do the early return below
+    const hasLeech = this.combatants.selected.leechRating > 0; // TODO replace when dynamic stats
+    let oneLeech = 0;
+    if(hasLeech && spellInfo.isLeech && !healVal.overheal) {
+      oneLeech = amount / this.combatants.selected.leechRating; // TODO replace when dynamic stats
+    } else if(!hasLeech) {
+      // TODO this will be a pain to implement
+    }
+    this.totalOneLeech += oneLeech;
+
+    if(spellInfo.multiplier) {
+      return; // multiplier heals don't count for weights and they're adjusted out for total healing
+    }
+    this.totalAdjustedHealing += healVal.effective;
+    if(spellInfo.ignored) {
+      return; // ignored heals don't count for weights but they are part of adjusted total healing
+    }
+
+    if(healVal.overheal) {
+      return; // if a spell overheals, we know it couldn't have healed for more
+    }
+
+    const hotCount = target.activeBuffs()
+        .map(buffObj => buffObj.ability.guid)
+        .filter(buffId => getSpellInfo(buffId).masteryStack)
+        .length;
+
+    // INT //
```

Comment:
You could also use regions;

```
// region Intellect
[code]
// endregion
```

Reply:
Do those do anything special when inspecting code? Unfamiliar with them.

Reply:
Most decent editors make them collapsible. You could also add a block around the code by simply inserting { and } around it, also makes start and end easier to spot.

The best thing would be to extract to new methods.

Reply:
They originally were their own methods, but there was a lot of changing method signatures and these were also methods that were going to be called from exactly one place. I think I'll go with the regions, to be honest.

Reply:
`these were also methods that were going to be called from exactly one place` that is not a bad thing. Readability and maintainability are very important.

Reply:
I started separating them out into their own functions and it really made it worse because of all the fields needed to be passed, repeated calculation, etc. I've cleaned up the code in `_handleHeal` and made them regions, I think it looks much better now.